### PR TITLE
Removes Python 3.0, Notebook 5 mentions from contributor docs

### PR DESCRIPTION
**Pull request by JasonWeill**
Removes a couple of references to obsolete, end-of-life software, from our contributor docs: Python 3.0, Notebook 5.

Also makes a couple of small copy edits to the rest of the section.

### CODE REVIEW

File: docs/source/developer/contributing.rst [hunk 1]
```diff
@@ -351,15 +351,10 @@ Notes:
...
--  Some of the packages used in the development environment require
-   Python 3.0 or higher. If you encounter an ImportError during the
-   installation, make sure Python 3.0+ is installed. Also, try using the
-   Python 3.0+ version of ``pip`` or ``pip3 install -e .`` command to
-   install JupyterLab from the forked repository.
 -  If you see an error that says ``Call to 'pkg-config pixman-1 --libs'
    returned exit status 127 while in binding.gyp`` while running the
    ``pip install`` command above, you may be missing packages required
-   by ``canvas``. Please see `Installing Node.js and jlpm section`_
+   by ``canvas``. Please see the `"Installing Node.js and jlpm" section`_
```
Comment (krassowski) [inline docs/source/developer/contributing.rst:357]: This breaks the test:

```
/home/runner/work/jupyterlab/jupyterlab/docs/source/developer/contributing.rst:439: ERROR: Unknown target name: ""installing node.js and jlpm" section".
```

I think you will need to use either rename the explicit target in line 292 or use the [`:ref:` syntax](https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#the-ref-role).

**Review by fcollonval — APPROVED** [review]
Thanks @JasonWeill

**Comment by krassowski** [comment]
I think the `header-license-fix` is failing because this branch is far behind `main`, I will click "Update branch" button to confirm it passes on merge.

**Review by krassowski — APPROVED** [review]
Thank you @JasonWeill!

**Comment by krassowski** [comment]
@meeseeksdev please backport to 4.0.x

**Pull request merged**